### DISCUSSION TITLE
播放几分钟开始严重发热(M2 MacBook air)

### DISCUSSION THREAD

**Discussion by luistrong** [amll-dev/applemusic-like-lyrics#257] [General]
应该是歌词模糊引起cpu大占用, 能否加个关闭歌词模糊选项, 毕竟逐行滚动就已经够用了, 模糊了还看不清上下文

**Comment by MianSoft** [comment]
> 能否加个关闭歌词模糊选项

该功能已经实现, 在提交 https://github.com/Steve-xmh/applemusic-like-lyrics/commit/b246f8d5cdc64c1fff251906f081bee138eebe81: core: 改善歌词绘制性能
您可以按照下方的 GIF 进行开关
![2024-09-23-16-40](https://github.com/user-attachments/assets/7868d2ae-7de5-4041-a1c0-b0f21aaddc32)
👆 这里有一个 7.12 MB 的 GIF

**Comment by Steve-xmh** [comment]
> 应该是歌词模糊引起cpu大占用, 能否加个关闭歌词模糊选项, 毕竟逐行滚动就已经够用了, 模糊了还看不清上下文

歌词行模糊为 iPad Apple Music 中自带的效果，因此是默认启用且可配置的。

本人也有同款配置的 MacBook Air，因为是浏览器而非原生实现，以至于歌词的动态变化会不断产生浏览器布局的重排和重新计算布局（这个对于网页来说是非常消耗性能的），而这个即便是关掉任何可能的效果都很难缓解这个现象（除非放弃弹簧动画效果使用滚动视图，但是效果会大打折扣），因此除非有新的保证效果的情况下的解决方案否则很难解决这个问题。
